### PR TITLE
Use the build-tools container where it makes sense.

### PR DESCRIPTION
**Pull request by geeknoid**
- Use the build tools container for test, lint, and check-config
targets. These CI jobs are now prduced through our config generator
instead of being hand-written.

- Remove deprecated scripts whose functionality is subsumed by
the common files and the build-tools container.

### CODE REVIEW

File: Makefile.core.mk
Comment (howardjohn) [inline Makefile.core.mk:1]: no go modules? or is it just in the top level?

File: Makefile.overrides.mk
Comment (howardjohn) [inline Makefile.overrides.mk:16]: this will break a lot of the stuff relating to the prow cluster I think, as it won't mount the kubernetes context. Or those makefiles get the context on demand with gcloud tool which is not in the docker image.

It's possible the makefiles for prow stuff is completely separate and you need to be in the prow/ folder to run it and none of this applies

File: Makefile.core.mk
Comment (howardjohn) [inline Makefile.core.mk]: this isn't in the container right?

File: Makefile.core.mk
Comment (howardjohn) [inline Makefile.core.mk:33]: is goarch not set by the container?

**Review by howardjohn — COMMENTED** [review]
looks good! just a couple questions about turning container build on

File: Makefile.core.mk
Comment (geeknoid) [reply Makefile.core.mk:1]: All container-based builds use go modules, we turn that on container-wide.

File: Makefile.overrides.mk
Comment (geeknoid) [reply Makefile.overrides.mk:16]: This setting is only relevant for stuff you run from the command-line using make. Anything run from Bazel is unaffected, anything run form Prow is unaffected.

File: Makefile.core.mk
Comment (geeknoid) [reply Makefile.core.mk:33]: GOARCH is set by default to the architecture of the host within the top-level Makefile. So it needs to be overridden, otherwise this ends up produring a mac binary that it tries to run in the Linux VM.

File: Makefile.overrides.mk
Comment (howardjohn) [reply Makefile.overrides.mk:16]: I think there are one-off commands that are run though. under prow/makefile. It may not be included in the top level makefiles though, on which case this would not apply to it

**Comment by geeknoid** [comment]
The lint-check job failure is transient, it'll go away after this change is merged.

File: Makefile.core.mk
Comment (geeknoid) [reply Makefile.core.mk]: Correct. What is configurator and where does it come from?

File: Makefile.overrides.mk
Comment (geeknoid) [reply Makefile.overrides.mk:16]: Yeah, I don't think that applies. This change only impacts build targets in Makefile.core.mk or anything it imports. Other makefiles are unrelated (although we'll eventually want to merge ther functionality as well so that all build activity can be automated and can be containerized)

File: Makefile.core.mk
Comment (geeknoid) [reply Makefile.core.mk]: I'm guessing this is the source for this tool:

https://github.com/kubernetes/test-infra/tree/master/testgrid/cmd/configurator

Comment (geeknoid) [reply Makefile.core.mk]: I've updated this to build the tool on the fly, but it's not quite working yet. Not clear why...

Comment (geeknoid) [reply Makefile.core.mk]: Works now. We could leave it this way, or we could put configurator into the build container. Not sure if its worth it. Thoughts?

**Review by howardjohn — APPROVED** [review]

File: Makefile.core.mk
Comment (fejta) [inline Makefile.core.mk:36]: Why do we need to force the architecture? Seems like this won't work for mac?

**Review by fejta — APPROVED** [review]
/hold

File: Makefile.core.mk
Comment (geeknoid) [reply Makefile.core.mk:36]: We're building from a container. But we have a (I expect) temporary situation where we force the architecture of stuff built within the container to be the architecture of the host system. In this case, we need to build the binary in the container and run that binary, so we need to force the OS/ARCH to what the container needs.

We also want to support building outside of the container at which point this approach will not work. We'll need a few more variables to control this, but we're not there yet. So for now, we have a few spots across the repos where we force the OS/ARCH so that things work right in the container.

**Pull request merged**
